### ISSUE TITLE
src/ProjectTemplates/build.cmd doesn't install some packages

### ISSUE THREAD

**Issue by Tratcher** [aspnet/AspNetCore#7388]
From the old templating readme:
"You also need to get the packages these templates depend on into your package cache or else `dotnet new` restore will fail. The easiest way to get them to run is by letting the build run at least 1 test."

`src/ProjectTemplates/build.cmd` Does build the aspnetcore shared runtime and pre-install the packages it needs. However it does not install any of the other packages directly referenced by the templates.
https://github.com/aspnet/AspNetCore/blob/45ecfad074bf6abefd6d87b8bb18b94afaed4ede/src/ProjectTemplates/Web.ProjectTemplates/StarterWeb-CSharp.csproj.in#L20-L28
As such `dotnet new` gives the following error:
```
d:\temp\templates>dotnet new razor
The template "ASP.NET Core Web App" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/aspnetcore-template-3pn-210 for details.

Processing post-creation actions...
Running 'dotnet restore' on d:\temp\templates\templates.csproj...
  Restoring packages for d:\temp\templates\templates.csproj...
d:\temp\templates\templates.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.Mvc.NewtonsoftJson with version (>= 3.0.0-preview-t000)
d:\temp\templates\templates.csproj : error NU1102:   - Found 24 version(s) in DotNet [ Nearest version: 3.0.0-preview-19107-05 ]
d:\temp\templates\templates.csproj : error NU1102:   - Found 1 version(s) in nuget.org [ Nearest version: 3.0.0-preview-19075-0444 ]
d:\temp\templates\templates.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
d:\temp\templates\templates.csproj : error NU1102:   - Found 0 version(s) in https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
```

@natemcmaster

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg, can you please look into this? Thanks!

**Comment by natemcmaster** [comment]
Side note: it might be worth updating the templating tests to override the NUGET_PACKAGES env variable to a temp folder so that when you run tests, restore does not pollute your default package cache with 3.0.0-preview-t000 packages.

**Comment by ryanbrandenburg** [comment]
My recommended solution here is to [use the scripts](https://github.com/aspnet/AspNetCore/tree/master/src/ProjectTemplates/scripts) which handle all the (known) problems we run into because we're not using production builds. I already had some stuff to make that work better built into https://github.com/aspnet/AspNetCore/pull/8196, but if that PR isn't merged soon I can always split it out into a separate PR with less baggage.

**Comment by ryanbrandenburg** [comment]
@Tratcher does that answer the question/problem here?

**Comment by ryanbrandenburg** [comment]
Talked with @Tratcher offline, he says the failure listed here doesn't happen anymore but he's not sure if things work otherwise because he's blocked by problems within his code.

**Comment by ryanbrandenburg** [comment]
Closing this. @Tratcher, if you run into other/new problems file a new issue.
